### PR TITLE
Support offline mode for construction API

### LINKED ISSUE #77: Offline mode that only enables a subset of Construction API
All construction APIs now begin with a check of `ValidateNetworkIdentifier`. But according to [rosetta doc](https://www.rosetta-api.org/docs/flow.html#construction-api), most of construction APIs should be working offline. 

And we need them to work without `oasisClient` to integrate this project into bitpie wallet. Could you make this `oasisClient` in construction API service optional?

### PR DESCRIPTION
**Pull request by abukosek**
Closes #77.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,23 @@ port that you want the gateway to listen on (default is 8080).
...
 
+## Offline mode
+
+The gateway supports an "offline" mode, which enables only a subset of the
+Construction API and nothing else, but doesn't require a connection to an
+Oasis node.
+
+To enable it, set the environment variable `OASIS_ROSETTA_GATEWAY_OFFLINE_MODE`
+to a non-empty value.  You must also set the environment variable
+`OASIS_ROSETTA_GATEWAY_OFFLINE_MODE_CHAIN_ID` to the chain ID of the network
```
Comment (kostko) [inline README.md:50]: Maybe this should be called something else than "chain ID" as in the documentation we call it the ["genesis document hash"](https://docs.oasis.dev/oasis-core/high-level-components/index/genesis).

File: main.go [hunk 2]
```diff
@@ -114,14 +160,23 @@ func main() {
 		os.Exit(1)
 	}
 
-	logger.Info("connected to Oasis node", "chain_context", cid)
+	var router http.Handler
+	if !offlineMode {
```
Comment (kostko) [inline main.go:164]: Use a switch instead.

File: services/construction.go [hunk 3]
```diff
@@ -152,7 +169,13 @@ func (s *constructionAPIService) ConstructionHash(
...
 	request *types.ConstructionHashRequest,
 ) (*types.TransactionIdentifierResponse, *types.Error) {
-	terr := ValidateNetworkIdentifier(ctx, s.oasisClient, request.NetworkIdentifier)
+	var terr *types.Error
+	if s.oasisClient == nil {
+		// Running in offline mode, get chain ID from env var.
+		terr = ValidateNetworkIdentifierWithChainID(os.Getenv(OfflineModeChainIDEnvVar), request.NetworkIdentifier)
+	} else {
+		terr = ValidateNetworkIdentifier(ctx, *s.oasisClient, request.NetworkIdentifier)
+	}
```
Comment (kostko) [inline services/construction.go:178]: This seems like it is done over and over again, can you move it to a function? E.g., maybe `ValidateNetworkIdentifier` could handle both cases?

File: README.md (hunk 1 above)
Comment (abukosek) [reply README.md]: I've renamed it to "genesis document hash" in the README, so it matches the current docs.

File: services/construction.go (hunk 3 above)
Comment (abukosek) [reply services/construction.go]: Good idea, I've changed the `ValidateNetworkIdentifier` function to handle both.

**Review by kostko — APPROVED** [review]

**Pull request merged**